internal/controllers/provisioningrequest_hwprovision.go

```diff
@@ -289,6 +383,28 @@ func (t *provisioningRequestReconcilerTask) handleRenderHardwareTemplate(ctx con
 			"failed to get the Hardware template from ConfigMap %s, err: %w", hwTemplateCmName, err)
 	}
 
+	// Check if the NodePool exists
+	ns := utils.GetHwMgrPluginNS()
+	exist, err := utils.DoesK8SResourceExist(ctx, t.client, clusterInstance.GetName(), ns, nodePool)
+	if err != nil {
+		return nodePool, fmt.Errorf("failed to get NodePool %s in namespace %s: %w", clusterInstance.GetName(), ns, err)
+	}
+
+	if exist {
+		// Verify the changes
+		changed, err := utils.CompareConfigMapWithNodePool(hwTemplateCm, nodePool, nodeGroup)
+		if err != nil {
+			return nodePool, utils.NewInputError("%w", err)
+		}
+		if changed && !utils.IsProvisioningStateFulfilled(t.object) {
+			return nodePool, utils.NewInputError("hardware template changes are not allowed until the cluster provisioning is fulfilled")
+		}
+	} else {
+		if err = utils.ValidateConfigMapFields(hwTemplateCm); err != nil {
```

Comment:
This validates the required fields in the hardware template ConfigMap prepared by CSP. It makes sense to include this as part of the cluster template validation, allowing CSP to address issues earlier